src/Main.elm

```diff
@@ -1,16 +1,265 @@
+            Ok nodes ->
+                let
+                    upEntry =
+                        if hasUp then
+                            [ viewUp ]
+                        else
+                            []
+                in
+                    List.append upEntry <| List.map viewNode nodes
+
+            Err msg ->
+                [ div [ class "alert alert-error" ] [ text msg ] ]
+
+
+viewUp : Html Msg
+viewUp =
+    viewFolderEntry "Go Up" "go-up" (GoUp 1)
+
+
+viewNode : H.Node -> Html Msg
+viewNode node =
+    case node of
+        H.File f ->
+            a
+                [ href "#"
+                , class "audience"
+                , onClick (Select node)
+                ]
+                [ div [ class "panel panel-default" ]
+                    [ text f.name ]
+                ]
+
+        H.Folder count f ->
+            viewFolderEntry
+                (f.name ++ " (" ++ (toString count) ++ ")")
+                ""
+                (GoDown f.id)
+
+
+viewFolderEntry : String -> String -> Msg -> Html Msg
+viewFolderEntry label extraClasses onPickMsg =
+    a
+        [ href "#"
+        , class ("folder " ++ extraClasses)
+        , onClick onPickMsg
+        ]
+        [ div [ class "panel panel-default" ]
+            [ span
+                [ class "glyphicon glyphicon-folder-close" ]
+                []
+            , text label
+            ]
+        ]
+
+
+viewHierarchyInfo : Maybe H.Node -> Hierarchy -> Html msg
+viewHierarchyInfo s h =
+    div [ class "row" ]
+        [ div [ class "col-xs-12" ]
+            [ viewBreadcrumb h ]
+        , div [ class "col-xs-12" ]
+            [ h6 [] [ text "Selected Node" ] ]
+        , div [ class "col-xs-12" ]
+            [ pre [] [ text <| toString s ] ]
+        , div [ class "col-xs-12" ]
+            [ h6 [] [ text "Current path" ] ]
+        , div [ class "col-xs-12" ]
+            [ pre [] [ text <| toString <| H.currentPath h ] ]
+        ]
+
+
+viewBreadcrumb : Hierarchy -> Html msg
+viewBreadcrumb h =
+    let
+        history =
+            H.currentPath h
+    in
+        case history of
+            Ok path ->
+                ol [ class "breadcrumb" ]
+                    ((li [] [ a [ href "#" ] [ text "Root" ] ])
+                        :: (List.map viewBreadcrumbItem path)
+                    )
+
+            Err s ->
+                div [ class "alert alert-info", attribute "role" "alert" ]
+                    [ text s ]
+
+
+viewBreadcrumbItem : H.Node -> Html msg
+viewBreadcrumbItem node =
+    let
+        name =
+            case node of
+                H.File f ->
+                    f.name
+
+                H.Folder _ f ->
+                    f.name
+    in
+        li [] [ a [ href "#" ] [ text name ] ]
 
 
 {-| Main file of application
 -}
-main : Html msg
+main : Program Never
 main =
-    Html.text "There will be app soon!"
+    Html.beginnerProgram
+        { model = init
+        , view = view
+        , update = update
+        }
```

Comment:
I really appreciate that you managed to keep `Main` so simple and easy to understand. Giving you few more of these 👍 👍 👍  again.